internal/controllers/topology/cluster/desired_state.go

```diff
@@ -1345,21 +1345,24 @@ func templateToTemplate(in templateToInput) (*unstructured.Unstructured, error)
 	return template, nil
 }
 
-func ownerReferenceTo(obj client.Object) *metav1.OwnerReference {
+// ownerReferenceTo converts an object to an OwnerReference.
+// Note: We pass in gvk explicitly as we can't rely on GVK being set on all objects
+// (only on Unstructured).
+func ownerReferenceTo(obj client.Object, gvk schema.GroupVersionKind) *metav1.OwnerReference {
```

Comment:
should this be 
```suggestion
func ownerReferenceTo(gvk schema.GroupVersionKind, obj client.Object) *metav1.OwnerReference {
```

Reply:
I'm fine either way. I would probably put the object first as that is sort of the "main" parameter.